doc/source/data/transforming-data.rst

```diff
@@ -0,0 +1,284 @@
+.. _transforming_data:
+
+=================
+Transforming Data
+=================
+
+Datastream transforms take in datastreams and produce new datastreams. For example, *map_batches*
+is a transform that applies a
+:ref:`user-defined function <transform_datastreams_writing_udfs>` on each data record
+and returns a new datastream as the result. Datastream transforms can be composed to
+express a chain of computations.
+
+--------
+Overview
+--------
+
+There are two main types of supported transforms, summarized in the below table:
+
+* One-to-one: each input block will contribute to only one output
+  block, such as :meth:`ds.map_batches() <ray.data.Datastream.map_batches>`.
+* All-to-all: input blocks can contribute to multiple output blocks,
+  such as :meth:`ds.random_shuffle() <ray.data.Datastream.random_shuffle>`.
```

Comment:
I feel like we should move this later in the guide or remove it altogether. I'm not sure if the reader actually needs to know this

Reply:
Agreed.